tests/unittests/tests-fmt/tests-fmt.c

```diff
@@ -239,11 +239,11 @@ static void test_fmt_s16_dfp(void)
     TEST_ASSERT_EQUAL_STRING("-12345", (char *)out);
 
     val = 31987;
-    fpp = 5;
+    fpp = 6;
     len = fmt_s16_dfp(out, val, fpp);
     out[len] = '\0';
-    TEST_ASSERT_EQUAL_INT(0, len);
```

Comment:
what was the original intent of this test? it seems like it only tests the same thing as the test on line 232 after this change.

Reply:
previously `fpp` needed to be `< 5` hence this test should return length `0` because of invalid `fpp`.

Reply:
If this does not test anything new, then lets remove this test like you did with the test on L304

Reply:
?? Its removed, or better it now tests something different, i.e., adding leading zeros, which wasn't possible before

Reply:
the function cannot return `0` anymore because I added an assert instead.

Reply:
or more correctly, it can return `0` because its a perfectly normal number, but it does not return `0` on error.